src/org/zaproxy/zap/extension/formhandler/PopupDialogAddField.java

```diff
@@ -0,0 +1,208 @@
+		super(owner, DIALOG_TITLE);
+	}
+
+	protected PopupDialogAddField(Dialog owner, String title) {
+		super(owner, title);
+	}
+
+	protected PopupDialogAddField(Dialog owner, String name, String[] values) {
+		super(owner, DIALOG_TITLE);
+		getNameTextField().setText(name);
+		for (String value : values) {
+			getValueField().addItem(value);
+		}
+	}
+
+	@Override
+	protected JPanel getFieldsPanel() {
+		JPanel fieldsPanel = new JPanel();
+
+		GroupLayout layout = new GroupLayout(fieldsPanel);
+		fieldsPanel.setLayout(layout);
+		layout.setAutoCreateGaps(true);
+		layout.setAutoCreateContainerGaps(true);
+
+		JLabel nameLabel = new JLabel(NAME_FIELD_LABEL);
+		JLabel valueLabel = new JLabel(VALUE_FIELD_LABEL);
+		JLabel enabledLabel = new JLabel(ENABLED_FIELD_LABEL);
+
+		layout.setHorizontalGroup(layout.createSequentialGroup()
+				.addGroup(layout.createParallelGroup(GroupLayout.Alignment.TRAILING).addComponent(nameLabel)
+						.addComponent(valueLabel).addComponent(enabledLabel))
+				.addGroup(layout.createParallelGroup(GroupLayout.Alignment.LEADING).addComponent(getNameTextField())
+						.addComponent(getValueField()).addComponent(getEnabledCheckBox())));
+
+		layout.setVerticalGroup(layout.createSequentialGroup()
+				.addGroup(layout.createParallelGroup(GroupLayout.Alignment.BASELINE).addComponent(nameLabel)
+						.addComponent(getNameTextField()))
+				.addGroup(layout.createParallelGroup(GroupLayout.Alignment.BASELINE).addComponent(valueLabel)
+						.addComponent(getValueField()))
+				.addGroup(layout.createParallelGroup(GroupLayout.Alignment.BASELINE).addComponent(enabledLabel)
+						.addComponent(getEnabledCheckBox())));
+
+		return fieldsPanel;
+	}
+
+	@Override
+	protected String getConfirmButtonLabel() {
+		return CONFIRM_BUTTON_LABEL;
+	}
+
+	@Override
+	protected void init() {
+		getEnabledCheckBox();
+		this.pack();
+	}
+
+	/*
+	 * Confirms the fields do not already exist
+	 */
+	@Override
+	protected boolean validateFields() {
+		String fieldName = getNameTextField().getText().toLowerCase();
+		for (String field : Control.getSingleton().getExtensionLoader().getExtension(ExtensionFormHandler.class)
+				.getFormHandlerFieldNames()) {
+			if (fieldName.equals(field)) {
+				JOptionPane.showMessageDialog(this, TEXT_NAME_REPEATED_DIALOG, TITLE_NAME_REPEATED_DIALOG,
+						JOptionPane.INFORMATION_MESSAGE);
+				getNameTextField().requestFocusInWindow();
+				return false;
+			}
+		}
+
+		return true;
+	}
+
+	/**
+	 * When the Add button is clicked, create a new field. This field will be
+	 * created with the name, value and enabled input by the user. The name will
+	 * always be lower case.
+	 */
+	@Override
+	protected void performAction() {
+		extFormHandler.addFormHandlerFieldName(getNameTextField().getText().toLowerCase(),
+				getValueField().getEditor().getItem().toString());
+	}
+
+	@Override
+	protected void clearFields() {
+		getNameTextField().setText("");
+		getNameTextField().discardAllEdits();
```

Comment:
Since the JComboBox for the value is editable do I need to discard edits for it too?

Reply:
No, that class (or its editor) does not keep track of edits.